src/test/java/de/tum/in/www1/artemis/CourseIntegrationTest.java

```diff
@@ -124,6 +124,101 @@ public void testCreateCourseWithPermission() throws Exception {
         assertThat(courseRepo.findAll()).as("Course has not been stored").contains(repoContent.toArray(new Course[0]));
     }
 
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void testCreateCourseWithSameShortName() throws Exception {
+        Course course = ModelFactory.generateCourse(null, null, null, new HashSet<>());
+        course.setShortName("shortName");
+        jiraRequestMockProvider.enableMockingOfRequests();
+        jiraRequestMockProvider.mockCreateGroup(course.getDefaultStudentGroupName());
+        jiraRequestMockProvider.mockCreateGroup(course.getDefaultTeachingAssistantGroupName());
+        jiraRequestMockProvider.mockCreateGroup(course.getDefaultInstructorGroupName());
+        request.post("/api/courses", course, HttpStatus.CREATED);
+        List<Course> repoContent = courseRepo.findAll();
+        assertThat(repoContent.size()).as("Course got stored").isEqualTo(1);
+        course = ModelFactory.generateCourse(null, null, null, new HashSet<>());
+        course.setShortName("shortName");
+        request.post("/api/courses", course, HttpStatus.BAD_REQUEST);
+        assertThat(courseRepo.findAll()).as("Course has not been stored").contains(repoContent.toArray(new Course[0]));
```

Comment:
If I'm correct, this tests if the old course still exists, but not of the new one does not exist.

Reply:
thank you for the review, i adjusted the test case